### PR TITLE
Implement Staking Pool Contract

### PR DESCRIPTION
**Pull request by wild-mouse**
- Implement `StakingPool` contract
- Update `StakingPool#earned` function interface for mock arguments
- Add minter variable into `FanToken` contract to restrict minting
- Put TODO comment such as adding events

### CODE REVIEW

File: contract/contracts/StakingPool.sol [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    mapping(address  => uint256) public tokensTotalSupply;
+    mapping(address => mapping(address => uint256)) private creatorsTokenBalances;
+    mapping(address => bool) public registeredTokens;
+
+    function balanceOf(address account, address token) public view returns(uint256){
+        return creatorsTokenBalances[account][token];
+    }
+
+    function addTokenToStakingList(address token) public onlyOwner {
+        registeredTokens[token] = true;
```
Comment (toshiakitakase) [inline contract/contracts/StakingPool.sol:23]: I think we've made the specification to allow staking to be stopped, so it should be changeable to false.

Comment (toshiakitakase) [reply contract/contracts/StakingPool.sol:23]: And I think the creator should be the one who can change that state.

File: contract/contracts/StakingPool.sol [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        uint8 decimals
+    ) public override view returns(uint256) {
+        // TODO: claim amount is currently simplest mock. It will be replaced with formula
+        return balanceOf(account, token).mul(10 ** uint256(decimals)).div(totalSupply);
+    }
+
+    function stake(uint256 amount, address token) public override {
+        creatorsTokenBalances[msg.sender][token] = creatorsTokenBalances[msg.sender][token].add(amount);
+        tokensTotalSupply[token] = tokensTotalSupply[token].add(amount);
+        require(registeredTokens[token], "Token is not registered to staking list");
```
Comment (toshiakitakase) [inline contract/contracts/StakingPool.sol:39]: To increase gas efficiency, it is friendly to write the requirement on the first line.

**Review by toshiakitakase — APPROVED** [review]

**Comment by toshiakitakase** [comment]
LGTM. Everything else is fine.

File: contract/contracts/StakingPool.sol (hunk 1 above)
Comment (wild-mouse) [reply contract/contracts/StakingPool.sol:23]: You are right. I'm gonna fix it in other PR.

**Pull request merged**